lib/ubi_cli.rb

```diff
@@ -66,6 +66,14 @@ def self.list(cmd, label, fields, fragment: cmd)
     end
   end
 
+  def self.destroy(cmd, label, fragment: cmd)
+    UbiRodish.on(cmd).run_is("destroy") do
+      delete(project_path("location/#{@location}/#{fragment}/#{@name}")) do |_, res|
+        ["#{label}, if it exists, is now scheduled for destruction"]
+      end
```

Comment:
for some later time, often, for destructive operations, many CLIs will prompt or ask you to type something.

Reply:
I thought about this, but the API doesn't require this, so I don't think the CLI should either.  We could potential added a `-f`/`--force` option, and have the CLI call fail without the option, but I don't like that.  It would be preferable to making the CLI interactive, though.  In general, I think we should assume that CLI users know what they are doing (or at least are OK with taking responsibility for their CLI use).

Reply:
I don't think the API-therefore-CLI syllogism holds. An API can't prompt, it's only an interface, a CLI exists to do things an API can't really do, since an API arguably defines itself in being abstract from a client implementation, it's something of a category error.

The problem is, it comes back as desperate support tickets for which varying surgeries are possible to fix the customer. Here are two kinds of errors:

* hacking up a previous command and bouncing on the enter key.
* accidentally recalling a previous dangerous command with ctrl-r, and bouncing on the enter key

Reply:
Well, I'm happy to make it work the way you want, you just need to describe how you want it to work.  Making `ubi` interactive (prompting the user for input) will require changes to the `ubi` program, and the main advantage of the thin-CLI approach is you need do not to make changes to the program to add features to the CLI. I'd prefer to make any necessary changes to `ubi` earlier rather than later.

Reply:
@geemus , @byucesoy : do you have opinions on re-prompts in the CLI? The former, as a general principle from experience, the latter, because you might be on the hook for "oops" (and yet others we simply have to say "SOL")

IIRC heroku had a `--confirm` flag for non-interactive mode, once you used it, you knew you were living dangerously, so often I wouldn't unless all the identifiers were safe even in unintended contexts/times of execution. But otherwise I believe it re-prompted you for the identifier name, which would stop a wild paste or keyboard bounce. I guess I didn't have a problem with this, but I never gave it much thought, either.

Reply:
I was planning to write similar comments in the #cli channel, but it seems like it has already been discussed here. I think it should ask for confirmation by default for destructive operations, and `--force` should bypass that.

I agree that CLI users know what they are doing, but people might accidentally run destructive commands on the wrong resource due to typos or selecting the wrong command from CTRL+R, etc. Once executed, there's no way to cancel or revert it. IMHO, a confirmation feels safer.

In the future, we might add a `production` or `do not allow destruction` toggle in the UI to help prevent accidental deletions of production resources.

Reply:
I'll work on support for confirmation for `destroy` commands.  Are there any other commands that should require confirmation?

Reply:
I think `destroy` is enough for now. May be resetting super user password 🤔

Reply:
I looked at that one, but we don't require confirmation on the web for it, so it doesn't make sense to require confirmation in the cli.